### PR TITLE
Add android.permission.WAKE_LOCK permission to abstract_method_smoke_test

### PR DESCRIPTION
**Pull request by blasten**
Issue: https://github.com/flutter/flutter/issues/42349
Log: https://api.cirrus-ci.com/v1/task/5683028134723584/logs/test.log

### CODE REVIEW

**Review by dnfield — CHANGES_REQUESTED** [review]
Could we instead just add the WAKE_LOCK permission tot he AndroidManifest for the affected app?

**Comment by blasten** [comment]
@dnfield sure.  The plugin `google_maps_flutter` doesn't indicate that it needs this permission though:  https://pub.dev/packages/google_maps_flutter . Should this be fixed there? 

cc @iskakaushik who owns this plugin.

**Comment by dnfield** [comment]
I can't find any evidence that a Map project in Android requires this permission, and a quick look at the Java source for the plugin doesn't seem to need it either.

File: dev/integration_tests/abstract_method_smoke_test/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.example.abstract_method_smoke_test">
-
+    <uses-permission android:name="android.permission.WAKE_LOCK"/>
```
Comment (dnfield) [inline dev/integration_tests/abstract_method_smoke_test/android/app/src/main/AndroidManifest.xml]: Add a TODO to remove this or figure out why it's needed, with a link to a bug for it.

**Comment by dnfield** [comment]
The stack trace is suggesting this has to do with platform view code, which may or may not be due to the map.  Maybe resizing this specific platform view triggers some weird code?

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
I think we should land this on red since it's fixing a flake that's part of the cause of redness.

**Comment by chinmaygarde** [comment]
Redness is not a flake. It is an issue to be fixed by https://github.com/flutter/flutter/pull/42350

**Comment by dnfield** [comment]
Yes- but we don't want the fix to go red because of this flake :)

**Pull request merged**